ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/QueryAnonymizer.java

```diff
@@ -877,30 +877,42 @@ private static String visitWithinExpression(final WithinExpressionContext contex
         final SqlBaseParser.SingleJoinWindowContext singleWithin
             = (SqlBaseParser.SingleJoinWindowContext) context;
 
-        stringBuilder.append(getSizeAndUnitFromJoinWindowSize(singleWithin.joinWindowSize()));
+        stringBuilder.append(String.format("%s",
+            anonymizeJoinWindowSize(singleWithin.joinWindowSize())));
       } else if (context instanceof JoinWindowWithBeforeAndAfterContext) {
         final SqlBaseParser.JoinWindowWithBeforeAndAfterContext beforeAndAfterJoinWindow
             = (SqlBaseParser.JoinWindowWithBeforeAndAfterContext) context;
 
-        final String before =
-            getSizeAndUnitFromJoinWindowSize(beforeAndAfterJoinWindow.joinWindowSize(0));
-        final String after =
-            getSizeAndUnitFromJoinWindowSize(beforeAndAfterJoinWindow.joinWindowSize(1));
-        stringBuilder.append(String.format("(%s, %s)", before, after));
+        stringBuilder.append(String.format("(%s, %s)",
+            anonymizeJoinWindowSize(beforeAndAfterJoinWindow.joinWindowSize(0)),
+            anonymizeJoinWindowSize(beforeAndAfterJoinWindow.joinWindowSize(1))));
+      } else if (context instanceof SqlBaseParser.JoinWindowWithGraceContext) {
+        final SqlBaseParser.JoinWindowWithGraceContext joinWithGracePeriod =
+            (SqlBaseParser.JoinWindowWithGraceContext) context;
+
+        stringBuilder.append(String.format("(SIZE %s, GRACE PERIOD %s)",
+            anonymizeJoinWindowSize(joinWithGracePeriod.joinWindowSize()),
+            anonymizeGracePeriod(joinWithGracePeriod.gracePeriodClause())));
       } else {
         throw new RuntimeException("Expecting either a single join window, ie \"WITHIN 10 "
-            + "seconds\", or a join window with before and after specified, "
-            + "ie. \"WITHIN (10 seconds, 20 seconds)");
+            + "seconds\", a join window with before and after specified, "
+            + "ie. \"WITHIN (10 seconds, 20 seconds)\", or a join window with a grace period, "
```

Comment:
Just realizing that this error message does not match our docs:
```
 [ WITHIN [(before TIMEUNIT, after TIMEUNIT) | N TIMEUNIT] ]
```

But the error message aligns to the grammar definition. Something seems to be off?